src/transformState.js

```diff
@@ -4,8 +4,44 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties': {
+        const { extraData } = action;
+
+        for (const [key, value] of Object.entries(extraData)) {
+          state[key] = value;
+        }
+        break;
+      }
+
+      case 'removeProperties': {
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
```

Comment:
it's better to create separate function for this, so our `switch-case` would e smaller and more readable